### PR TITLE
Python pre-stack support

### PR DESCRIPTION
**Pull request by jokva**
Will be rebased once https://github.com/Statoil/SegyIO/pull/56 is merged.

Initial prestack support in python, currently without documentation. Can be reviewed as if it was done, but expect some more commits before this is ready for merging.

### CODE REVIEW

**Comment by jokva** [comment]
This is more or less done, but https://github.com/Statoil/SegyIO/pull/56 should be merged first and this rebased.

File: README.md [hunk 1]
```diff
@@ -107,10 +107,11 @@ Small SEG Y formatted files are included in the repository for test purposes.
 Phyiscally speaking the data is non-sensical, but it is reproducible by using
 Segyio. The tests file are located in the tests/test-data directory. To
 reproduce the data file, build Segyio and run the test program `make-file.py`
-as such:
+and make-ps-file as such:
```
Comment (jepebe) [inline README.md]: You should quote `make-ps-file.py` like `make-file.py`

File: examples/make-ps-file.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                           "[first xline] [last xline]",
+                           "([first offset] [last offset])"
+                         ).format(sys.argv[0]))
+
+    spec = segyio.spec()
+    filename = sys.argv[1]
+
+# to create a file from nothing, we need to tell segyio about the structure of
+# the file, i.e. its inline numbers, crossline numbers, etc. You can also add
+# more structural information, but offsets etc. have sensible defautls. This is
```
Comment (jepebe) [inline examples/make-ps-file.py]: defautls?

File: README.md (hunk 1 above)
Comment (jokva) [inline README.md]: Good catch!

Comment (jokva) [inline README.md]: https://github.com/Statoil/SegyIO/pull/56 Fix'd here

File: examples/make-ps-file.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        trace = np.arange(start = start,
+                          stop  = start + step * spec.samples,
+                          step  = step,
+                          dtype = np.single)
+
+        # one inline is N traces concatenated. We fill in the xline number
+        line = np.concatenate([trace + (xl / 100.0) for xl in spec.xlines])
+        line = line.reshape( (len(spec.xlines), spec.samples) )
+
+        xlines, offsets  = len(spec.xlines), len(spec.offsets)
```
Comment (kjellkongsvik) [inline examples/make-ps-file.py]: xlines vs spec.xlines have different meaning. should be more explicit. e.g. xlines_count or something.

File: python/segyio/segy.py [hunk 4]
```diff
@@ -393,6 +406,15 @@ def iline(self):
         an error. Note that each line is returned as a numpy array, meaning
         accessing the intersections of the inline and crossline is 0-indexed.
 
+        Additionally, the iline mode has a concept of offsets, which is useful
```
Comment (kjellkongsvik) [inline python/segyio/segy.py]: iline mode -> line mode

File: examples/make-ps-file.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    with segyio.create(filename, spec) as f:
+        # one inline consists of 50 traces
+        # which in turn consists of 2000 samples
+        start = 0.0
+        step = 0.00001
+        # fill a trace with predictable values: left-of-comma is the inline
+        # number. Immediately right of comma is the crossline number
+        # the rightmost digits is the index of the sample in that trace meaning
+        # looking up an inline's i's jth crosslines' k should be roughly equal
+        # to i.j0k. Offset values are multiplied by 100
```
Comment (kjellkongsvik) [inline examples/make-ps-file.py]: add offset to formula i.j0k, e.g. o0i.j0k?

Comment (jokva) [inline examples/make-ps-file.py]: Done.

**Pull request merged**